src/test/scala/mesosphere/marathon/state/GroupManagerTest.scala

```diff
@@ -87,7 +87,80 @@ class GroupManagerTest extends MarathonActorSupport with MockitoSugar with Match
     ))
     val update = manager(minServicePort = 10, maxServicePort = 20).assignDynamicServicePorts(Group.empty, group)
     update.transitiveApps.filter(_.hasDynamicPort) should be ('empty)
-    update.transitiveApps.flatMap(_.portNumbers.filter(x => x >= 10 && x <= 20)) should have size 2
+    update.transitiveApps.flatMap(_.hostPorts.flatten.filter(x => x >= 10 && x <= 20)) should have size 2
+  }
+
+  test("Assign dynamic service ports specified in multiple containers") {
+    import Container.Docker
+    import Docker.PortMapping
+    import org.apache.mesos.Protos.ContainerInfo.DockerInfo.Network
+    val c1 = Some(Container(
+      docker = Some(Docker(
+        image = "busybox",
+        network = Some(Network.USER),
+        portMappings = Some(Seq(
+          PortMapping(containerPort = 8080)
+        ))
+      ))
+    ))
+    val c2 = Some(Container(
+      docker = Some(Docker(
+        image = "busybox",
+        network = Some(Network.USER),
+        portMappings = Some(Seq(
+          PortMapping(containerPort = 8081)
+        ))
+      ))
+    ))
+    val group = Group(PathId.empty, Set(
+      AppDefinition("/app1".toPath, portDefinitions = Seq(), container = c1),
+      AppDefinition("/app2".toPath, portDefinitions = Seq(), container = c2)
+    ))
+    val update = manager(minServicePort = 10, maxServicePort = 20).assignDynamicServicePorts(Group.empty, group)
+    update.transitiveApps.filter(_.hasDynamicPort) should be ('empty)
```

Comment:
what's the difference? I just copy/pasted

On Wed, Jun 22, 2016 at 7:47 AM, Gastón Kleiman <notifications@github.com>
wrote:

> In src/test/scala/mesosphere/marathon/state/GroupManagerTest.scala
> <https://github.com/mesosphere/marathon/pull/4008#discussion_r68038551>:
>
> > +    ))
> > +    val c2 = Some(Container(
> > +      docker = Some(Docker(
> > +        image = "busybox",
> > +        network = Some(Network.USER),
> > +        portMappings = Some(Seq(
> > +          PortMapping(containerPort = 8081)
> > +        ))
> > +      ))
> > +    ))
> > +    val group = Group(PathId.empty, Set(
> > +      AppDefinition("/app1".toPath, portDefinitions = Seq(), container = c1),
> > +      AppDefinition("/app2".toPath, portDefinitions = Seq(), container = c2)
> > +    ))
> > +    val update = manager(minServicePort = 10, maxServicePort = 20).assignDynamicServicePorts(Group.empty, group)
> > +    update.transitiveApps.filter(_.hasDynamicPort) should be ('empty)
>
> You can use empty instead of 'empty
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4008/files/f1d2aa955451bae801682eed5c016afadcfc91eb#r68038551>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACPVLF0FgbigY48R9vHQosSresw8yhN_ks5qOSDKgaJpZM4I6Oac>
> .
>